browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc

```diff
@@ -83,6 +86,10 @@ void BraveNewTabPageHandler::InitForSearchPromotion() {
                           base::Unretained(this)));
   template_url_service_observation_.Observe(
       TemplateURLServiceFactory::GetForProfile(profile_));
+
+  brave_search_conversion::p3a::RecordPromoShown(
```

Comment:
This is called whenever NTP is created if promotion is enabled.
However, NTP will not show promotion when sponsored images are displayed.
When sponsored images are displayed, this should not be called.
I'll push another commit.

Reply:
Pushed https://github.com/brave/brave-core/pull/14189/commits/0e00a36fca72a75a57773c5e6defe7e1496104c6